modules/aql/arrow-query-language/src/main/kotlin/arrow/aql/Union.kt

```diff
@@ -0,0 +1,23 @@
+package arrow.aql
+
+import arrow.core.identity
+import arrow.data.ForListK
+import arrow.data.ListK
+import arrow.data.k
+import arrow.instances.list.semigroupK.combineK
+import arrow.instances.listk.monoid.monoid
+import arrow.typeclasses.Foldable
+
+interface Union<F> {
+
+  fun foldable(): Foldable<F>
+
+  infix fun <A, B, Z> Query<F, A, Z>.union(query: Query<F, B, Z>): Query<ForListK, Z, Z> =
+    foldable().run {
+      val la: ListK<Z> = from.foldMap(ListK.monoid()) { listOf(select(it)).k() }
+      val lb: ListK<Z> = query.from.foldMap(ListK.monoid()) { listOf(query.select(it)).k() }
+      val result: ListK<Z> = la.combineK(lb).k()
```

Comment:
No need for `k()`. Could be just `val result: ListK<Z> = la.combineK(lb)`

Reply:
`result` has to be a Kinded value and `List` on its own isn't.